### ISSUE TITLE
extend `avoid_types_as_parameter_names` to include record fields

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#3628]

**Comment by pq** [comment]
**BAD:**

```dart
var c = (int: 1);
```

**GOOD:**

```dart
var c = (index: 1);
```
